code/game/objects/items/v8_engine.dm

```diff
@@ -91,15 +95,31 @@
 	if(fire_charges <= 0)
 		balloon_alert(user, "no fire charges!")
 		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
-	user.throw_at(target = get_turf(target), range = 2 * fire_charges, speed = 5, thrower = user, spin = FALSE, gentle = FALSE, quickstart = TRUE)
-	COOLDOWN_START(src, fire_charge_cooldown, DASH_COOLDOWN)
-	reset_charges(on_dash = TRUE)
+	var/saved_fire_charges = fire_charges
+	addtimer(CALLBACK(src, PROC_REF(teleport_dash), user), 0.5 SECONDS * saved_fire_charges)
+	if(user.dropItemToGround(src))
+		throw_at(target = target, range = 2 * saved_fire_charges, speed = 5, thrower = user, spin = TRUE, gentle = FALSE, quickstart = TRUE)
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
 /obj/item/house_edge/update_icon_state()
 	inhand_icon_state = HAS_TRAIT(src, TRAIT_WIELDED) ? "house_edge1" : "house_edge0"
 	return ..()
 
+/**
+ * Teleport dash for the House Edge. Teleports the user to the sword and puts the sword back in the user's hand.
+ */
+/obj/item/house_edge/proc/teleport_dash(mob/living/user)
+	if(QDELETED(src) || user.incapacitated())
+		return FALSE
+	if(isnull(user) || ismob(loc))
+		reset_charges(on_dash = TRUE)
+		return FALSE
+	new /obj/effect/temp_visual/fire_fizzle(get_turf(user))
+	user.forceMove(drop_location())
```

Comment:
Also you're using forcemove rather than doteleport which means this can bypass noteleport stuff